sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_client.py

```diff
@@ -354,8 +354,9 @@ def get_queue_receiver(
          will be immediately removed from the queue, and cannot be subsequently rejected or re-received if
          the client fails to process the message. The default receive_mode is PEEK_LOCK.
         :paramtype receive_mode: Union[~azure.servicebus.ServiceBusReceiveMode, str]
-        :keyword Optional[float] max_wait_time: The timeout in seconds between received messages after which the
-         receiver will automatically stop receiving. The default value is None, meaning no timeout. If connection
+        :keyword Optional[float] max_wait_time: The timeout in seconds to wait for the first message to arrive.
```

Comment:
do the `receive_messages` docs need to be updated too? also, are there any differences b/w the pull vs. push receive that we would want to point out here?

Reply:
for `receive_messages` I think mentioning that specifying a value there, would override the init.  I was also thinking if using samples with comments can help with understanding the nuances of using `max_wait_time` under different scenarios
* non - session based queue -> `receive_messages` vs iterator
* session based queue -> `receive_messages` vs iterator
* session based queue but `next_available_session` - > `receive_messages` vs iterator

Reply:
The updated description only address the "first message". I assume this timeout also plays a part in how long I will wait for subsequent messages?

Reply:
For receive_messages -- we wait max_wait_time to receive messages all at once and then return.
For the iterator -- we will wait max_wait_time to receive each message.

It might be best to change to back this and clarify with samples like Kashif mentioned:

The timeout in seconds to wait for messages to arrive.
         If no messages arrive, and no timeout is specified, this call will not return
         until the connection is closed. The default value is None, meaning no timeout. If connection
         errors are occurring due to write timing out, the connection timeout value may need to be adjusted. See
         the `socket_timeout` optional parameter for more details.